packages/headless/src/controllers/relevance-inspector/headless-relevance-inspector.ts

```diff
@@ -17,40 +30,150 @@ import {
   validateInitialState,
   validateOptions,
 } from '../../utils/validate-payload';
-import {buildController} from '../controller/headless-controller';
+import {buildController, Controller} from '../controller/headless-controller';
+
+export {
+  RankingInformation,
+  DocumentWeights,
+  TermWeightReport,
+  QueryRankingExpressionWeights,
+  ExecutionReport,
+  ExecutionStep,
+  UserIdentity,
+  RankingExpression,
+};
 
 export interface RelevanceInspectorProps {
+  /**
+   * The initial state that should be applied to the `RelevanceInspector` controller.
+   */
   initialState?: RelevanceInspectorInitialState;
+
+  /**
+   * The options for the `RelevanceInspector` controller.
+   */
   options?: RelevanceInspectorOptions;
 }
 
 const initialStateSchema = new Schema({
-  /**  If debug mode should be enabled */
   enabled: new BooleanValue({default: false}),
 });
 
 const optionsSchema = new Schema({
-  /** Whether to automatically log state to console on new responses */
   automaticallyLogInformation: new BooleanValue({default: true}),
 });
 
-export type RelevanceInspectorInitialState = SchemaValues<
-  typeof initialStateSchema
->;
-
-export type RelevanceInspectorOptions = SchemaValues<typeof optionsSchema>;
+export interface RelevanceInspectorInitialState {
+  /**
+   * Whether debug mode should be enabled.
+   * */
+  enabled?: boolean;
+}
 
-export type RelevanceInspectorState = RelevanceInspector['state'];
+export interface RelevanceInspectorOptions {
+  /**
+   * Whether to automatically log state to console on new responses.
+   * */
+  automaticallyLogInformation?: boolean;
+}
 
 /**
  * The `RelevanceInspector` controller is in charge of allowing displaying various debug information.
  */
-export type RelevanceInspector = ReturnType<typeof buildRelevanceInspector>;
+export interface RelevanceInspector extends Controller {
+  /**
+   * Enables debug mode.
+   */
+  enable(): void;
+
+  /**
+   * Disables debug mode.
+   */
+  disable(): void;
+
+  /**
+   * Logs debug information to the console.
+   */
+  logInformation(): void;
+
+  /**
+   * The state of the `RelevanceInspector` controller.
+   * */
+  state: RelevanceInspectorState;
+}
+
+export interface RelevanceInspectorState {
+  /**
+   * Whether debug mode is enabled.
+   * */
+  isEnabled: boolean;
+
+  /**
+   * The ranking information for every result.
+   */
+  rankingInformation?: ResultRankingInformation[];
+
+  /**
+   * The query execution report.
+   */
+  executionReport?: ExecutionReport;
 
+  /**
+   * The query expressions sent in the request.
+   */
+  expressions?: QueryExpressions;
+
+  /**
+   * The security identities.
+   */
+  userIdentities?: UserIdentity[];
```

Comment:
I kept the `UserIdentity` interface name, but considered renaming to `SecurityIdentity` since it could contain groups. What term is usually used to refer to these identities?

Reply:
You're right, we usually use the term 'security identity' to describe an identity of any type, so I think that `SecurityIdentity` would be more appropriate.

Reply:
Ok, will rename to `SecurityIdentity`.